stdlib/3/io.pyi

```diff
@@ -85,16 +84,20 @@ class BufferedIOBase(IOBase):
     def read(self, __size: Optional[int] = ...) -> bytes: ...
     def read1(self, __size: int = ...) -> bytes: ...
 
-class FileIO(RawIOBase):
+class FileIO(RawIOBase, BinaryIO):
     mode: str
-    name: Union[int, str]
+    # Technically this is whatever is passed in as file, either a str, a bytes, or an int.
+    name: Union[int, str]  # type: ignore
```

Comment:
Why not type it as `Union[str, bytes, int]` then?

Reply:
I didn't want to broaden the type because nobody has complained about it, it's not directly related to the goal of this PR, and there's some chance broadening the type could cause new errors for people using the attribute.

Technically the right solution is to make various IO classes (not just this one but also at least TextIOWrapper) generic over (str, bytes, int) and give `.name` that generic type. But making these classes generic has the cost that now all users with `--disallow-any-generics` on have to specify the type parameter. That doesn't seem worth it for a rarely used attribute that nobody has even opened an issue about.